xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp

```diff
@@ -1309,39 +1309,39 @@ void CRenderManager::PrepareNextRender()
   // internal buffers of driver, assume that driver lets us go one frame in advance
   double totalLatency = DVD_SEC_TO_TIME(m_displayLatency) - DVD_MSEC_TO_TIME(m_videoDelay) + 2* frametime;
 
-  double clocktime = frameOnScreen + totalLatency;
+  double renderPts = frameOnScreen + totalLatency;
 
-  bool next = clocktime >= m_Queue[m_queued.front()].pts;
+  bool next = renderPts >= m_Queue[m_queued.front()].pts;
 
   if (next)
   {
     // see if any future queued frames are already due
-    std::deque<int>::reverse_iterator curr, prev;
-    int idx;
-    curr = prev = m_queued.rbegin();
-    ++prev;
-    while (prev != m_queued.rend())
+    auto iter = m_queued.begin();
+    int idx = *iter;
+    ++iter;
+    while (iter != m_queued.end())
     {
-      if (clocktime > m_Queue[*prev].pts + 2 * frametime &&  // previous frame is late
-          clocktime > m_Queue[*curr].pts + frametime)        // selected frame due
+      // the slot for rendering in time is [pts .. (pts + frametime)]
+      // renderer/drivers have internal queues, being slightliy late here does not mean that
+      // we are really late. If we don't recover here, player will take action
+      if (renderPts < m_Queue[*iter].pts + 0.8 * frametime)
```

Comment:
shouldn't this be a ```>``` as before? Since your're trying to find the index of the first queued frame that you DON'T want to skip. Current logic seems wrong.